### PR TITLE
Tatau 75 create share value in the horovod

### PR DESCRIPTION
**Pull request by IgorKalaidjan**
Run docker from docker

### CODE REVIEW

File: tatau_core/utils/downloader.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def _download_from_ipfs(self, multihash: str, file_names: list):
             try:
                 os.symlink(target_path, link_path)
             except FileExistsError as ex:
-                if os.readlink(link_path) == target_path:
+                if True or os.readlink(link_path) == target_path:
```
Comment (27tech) [inline tatau_core/utils/downloader.py]: Fix it

File: tatau_core/utils/ssh.py [hunk 2]
```diff
@@ -2,25 +2,23 @@
...
-PermitEmptyPasswords no
-ChallengeResponseAuthentication no
-PasswordAuthentication no
-PermitUserEnvironment yes
-UsePrivilegeSeparation no
-HostKey {}
-AuthorizedKeysFile {}
-Port 12345
-"""
+# SSHD_CONFIG = """
```
Comment (27tech) [inline tatau_core/utils/ssh.py]: Remove commented code

File: verifier.yml [hunk 3]
```diff
@@ -10,7 +10,9 @@ services:
     # Support nvidia docker 2
     runtime: nvidia
     volumes:
+      - tatau-data:/tmp/tatau
       - ./verifier:/app/verifier:ro
+      - tatau-sock:/var/run/docker.sock
```
Comment (27tech) [inline verifier.yml]: Socket is not a volume, u should mount it directly

Comment (27tech) [inline verifier.yml]:
```suggestion
      - /var/run/docker.sock:/var/run/docker.sock
```

File: verifier.yml [hunk 4]
```diff
@@ -26,4 +28,8 @@ services:
       TATAU_RAVEN_DSN: $TATAU_RAVEN_DSN
       KEYS_ROOT: $KEYS_ROOT
       VERIFIER_ACCOUNT_ADDRESS: $VERIFIER_ACCOUNT_ADDRESS
-      NET: $NET
\ No newline at end of file
+      NET: $NET
+
+volumes:
+  tatau-data:
+  tatau-sock:
```
Comment (27tech) [inline verifier.yml]: Remove this volume

File: worker.cpu.yml [hunk 5]
```diff
@@ -24,3 +25,6 @@ services:
       KEYS_ROOT: $KEYS_ROOT
       WORKER_ACCOUNT_ADDRESS: $WORKER_ACCOUNT_ADDRESS
       NET: $NET
+
+volumes:
+  tatau-data:
```
Comment (27tech) [inline worker.cpu.yml]: Tatau volume should be declared only once at whole stack, do it in common.yml

File: hvd_worker.py [hunk 6]
```diff
@@ -27,7 +27,6 @@ def load_credentials(account_address_var_name):
 if __name__ == '__main__':
     try:
         index = '_{}'.format(sys.argv[1])
```
Comment (27tech) [inline hvd_worker.py]: Is this block required?

File: hvd_worker.py [hunk 7]
```diff
@@ -27,7 +27,6 @@ def load_credentials(account_address_var_name):
 if __name__ == '__main__':
     try:
```
Comment (27tech) [inline hvd_worker.py]:
```suggestion
```

File: hvd_worker.py [hunk 8]
```diff
@@ -27,7 +27,6 @@ def load_credentials(account_address_var_name):
...
         index = '_{}'.format(sys.argv[1])
-        settings.TATAU_STORAGE_BASE_DIR
     except IndexError:
```
Comment (27tech) [inline hvd_worker.py]:
```suggestion
```

File: hvd_worker.py [hunk 9]
```diff
@@ -27,7 +27,6 @@ def load_credentials(account_address_var_name):
...
         index = ''
```
Comment (27tech) [inline hvd_worker.py]:
```suggestion
```

File: hvd_worker.py (hunk 6 above)
Comment (27tech) [reply hvd_worker.py]:
```suggestion
```

File: verifier.yml (hunk 4 above)
Comment (27tech) [reply verifier.yml]:
```suggestion
```

File: tatau_core/sessions/docker.py [hunk 10]
```diff
@@ -25,21 +33,23 @@ def get_volumes(self):
...
-            'detach': True,
-            'user': 1000,
-            'working_dir': '/tatau',
-            'hostname': kwargs.get('name')
+            'detach': False,
+            'privileged': True,
+            'hostname': kwargs.get('name'),
+            'links': links,
+            'network': settings.DOCKER_NET,
+            'environment': ['RING=%s' % settings.RING_NAME]
```
[the remark this answers is not in the document]
Comment (Yulya) [reply tatau_core/sessions/docker.py]: Will remove, cause passing value does not works the way we need. Redid it with session value

File: tatau_core/sessions/hvd_train.py [hunk 11]
```diff
@@ -31,7 +31,7 @@ def __init__(self, task_assignment_id, hostname=None):
         self.db = DB()
         logger.debug('AAAAAAAAA' + hostname)
         self.encryption = Encryption()
-        rsa_pk = SSHConfig(ROOT_DIR).load_private_key(path_suffix=hostname)
+        rsa_pk = SSHConfig('/root').load_private_key(path_suffix=hostname)
```
[the remark this answers is not in the document]
Comment (Yulya) [reply tatau_core/sessions/hvd_train.py]: These are different values.
SSHConfig is initialized with value where to find or create .ssh dir.

When it is called in main or train - there should go /root see docker.py and volumes mapping. We can use $HOME here for root user, bit we explicitly hardcode it in docker.py. 

In process assignment SSHConfig is initialized with some directory where we can dump keys and than share this dir with horovod containers. Putting it to ROOT_DIR/.ssh seems to be ok

File: hvd_worker.py (hunk 7 above)
Comment (Yulya) [reply hvd_worker.py]: I don't understand this and following comments

File: tatau_core/node/producer/hvd_producer_node.py [hunk 12]
```diff
@@ -904,9 +904,9 @@ def _process_task_declaration(self, task_declaration: HvdTaskDeclaration):
             self._process_epoch_in_progress(task_declaration)
             return
 
-        if task_declaration.state == HvdTaskDeclaration.State.VERIFY_IN_PROGRESS:
-            self._process_verify_in_progress(task_declaration)
-            return
+        # if task_declaration.state == HvdTaskDeclaration.State.VERIFY_IN_PROGRESS:
+        #     self._process_verify_in_progress(task_declaration)
+        #     return
```
Comment (makar21) [inline tatau_core/node/producer/hvd_producer_node.py]: remove instead comment code

File: tatau_core/sessions/docker.py [hunk 13]
```diff
@@ -11,35 +13,39 @@
...
             },
-            self.base_dir: {
-                'bind': self.base_dir,
+            'tatau_core_tatau-data': {
+                'bind': '/tmp/tatau',
                 'mode': 'rw'
             },
-            self.storage_dir_path: {
-                'bind': self.storage_dir_path,
+            'tatau_core_tatau-ssh': {
```
Comment (makar21) [inline tatau_core/sessions/docker.py]: use COMPOSE_PROJECT_NAME instead hardcode of 'tatau_core'

File: tatau_core/sessions/hvd_train.py [hunk 14]
```diff
@@ -31,7 +33,7 @@ def __init__(self, task_assignment_id, hostname=None):
         self.db = DB()
         logger.debug('AAAAAAAAA' + hostname)
```
Comment (makar21) [inline tatau_core/sessions/hvd_train.py]: remove AAAAAAAAAA

File: tatau_core/sessions/session.py [hunk 15]
```diff
@@ -80,7 +80,7 @@ def uuid(self):
     def base_dir(self):
         session_dir = os.path.join(tempfile.gettempdir(), 'tatau', self.uuid, 'session')
         if not os.path.exists(session_dir):
-            os.makedirs(session_dir)
+            os.makedirs(session_dir, mode=0o777)
```
Comment (makar21) [inline tatau_core/sessions/session.py]: mode=0o777 - is default value

**Pull request merged**